core/src/components/range/range.tsx

```diff
@@ -902,10 +903,14 @@ const renderKnob = (
 
 const ratioToValue = (ratio: number, min: number, max: number, step: number): number => {
   let value = (max - min) * ratio;
+
   if (step > 0) {
+    // round to nearest multiple of step, then add min
     value = Math.round(value / step) * step + min;
   }
-  return clamp(min, value, max);
+  const clampedValue = clamp(min, value, max);
+
+  return roundToMaxDecimalPlaces(clampedValue, min, max, step);
```

Comment:
Out of curiosity, why did you opt for this approach instead of multiplying by a scaled `step` (and then dividing by the scale amount)?

For example: 

```typescript
let value = (max - min) * ratio;

const SCALE = 100;
const SCALED_STEP = step * SCALE;

if (step > 0) {
  // round to nearest multiple of step, then add min
  value = Math.round(value / step) * SCALED_STEP + min;
}

return clamp(min, value / SCALE, max);
```

Scaling the value ensures that you deal with integers to avoid JavaScript's floating point precision funkiness. As long as you scale down by the same amount you scaled up, you should get a nice neat decimal as a multiple of `step`.

Reply:
Because it wouldn't work for all numbers. I'd rather not set an arbitrary limit on how high or low the props can be, nor the precision they can have. For example, a `SCALE` of `10000` sets the bounds within 0.0001 < n < (MAX_INT / 10000), and you can't have more than 4 decimal places -- 5.12345 * 10000 = 51234.5.

Reply:
Ah yeah good point. Though I think the current approach may have some accuracy issues too since `toFixed` can sometimes yield funky results.

```html
<ion-range
  min="0.1"
  max="0.54321876549876532"
  value="0.2"
  step="0.001"
>
  <span slot="label">Label</span>
</ion-range>
```

Video sample:

https://user-images.githubusercontent.com/2721089/236343077-c6a0f1b4-cbb2-4ab4-a686-30bdb775fab0.mov

Is there a defined level of precision we support with `ion-range`?

Reply:
What's happening there is you've got a `max` with the same number of decimal places as the result with the rounding errors, and `toFixed` is rounding to as many as the `max`, so the error isn't trimmed. I would say that's okay, since we wouldn't be able to tell the difference between a rounding error result and something legitimate in a case like that. (While in your case, the only legitimate value with that precision is the actual `max`, if the step were 0.00000000000000001 instead, the two would definitely be indistinguishable.)

I think it's okay that this means we implicitly only support up to a certain level of precision, because the problem is with floating point math itself, not Ionic. If people are using precision levels _that high_, they're going to run into rounding error issues like this all _over_ the place. 🤭 (Or in other words, JS itself only supports up to a certain level of precision anyway, unless you start using third party data types meant for this sort of thing. And I'd say supporting those is out of scope for this fix 😆)

Reply:
Sounds good to me! Could we add a little bit of explanation about these limitations in the code so other devs are aware?